functions/handle-web-signup-role.js

```diff
@@ -31,76 +80,146 @@ app.post('/handleWebSignUpRole', async (req, res) => {
-
-    })
-    .catch( ( error ) => {
-      return res.status( unsucessfulResponse ).send({ 'error': error });
+    }).catch((error) => {
+      return res.status( unsucessfulResponse ).send({'error': error });
     });
   })
   .catch( (error) => {
-    console.error( error );
-    return res.status( unsucessfulResponse ).send({ 'error': error });
+    return res.status(unsucessfulResponse).send({ 'error': error});
+  });
+});
+
+// Gets all users in firebase
+app.post('/getUsers', async (req,res) => {
+  res = setResAttr(res,'*','POST','Content-Type, application/json');
+
+  // Global scope, no need to return in arrow func
+  var userList = [];
+
+  // https://firebase.google.com/docs/auth/admin/manage-users
+  const listAllUsers = (nextPageToken) => {
+    // Listen to 1000 users at a time
+    admin.auth(adminApp).listUsers(1000, nextPageToken)
+                                                    .then((listUsersResult) => {
+      listUsersResult.users.forEach((userRecord) => {
+        if (userRecord.customClaims && userRecord.customClaims !== null) {
+          if (userRecord.customClaims.hasOwnProperty(MANAGER_STR) || 
+              userRecord.customClaims.hasOwnProperty(ADMIN_STR)) {
+            userList.push(userRecord);
+          }
+        }
+      });
+      // Recurse for next 1000 users
+      if (listUsersResult.pageToken) {
+        listAllUsers(listUsersResult.pageToken);
+      }
+      else {
+        res.status(SUCCESSFUL_RES).send({'Users': userList}).end();
+      }
+    }).catch((error) => {
+      res.status(UNSUCCESSFUL_RES).send({'error': error}).end();
+    });
+  };
+  
+
+  // Check for valid user token
+  admin.auth(adminApp).verifyIdToken(req.body.token).then((decodedToken) => {
+    admin.auth(adminApp).getUser(decodedToken.uid).then((userRecord) => {
+      let claims = userRecord.customClaims;
+      if(claims !== null && (ADMIN_STR in claims && MANAGER_STR in claims)) {
+        listAllUsers();
+      }
+    }).catch((error) => {
+      return res.status(UNSUCCESSFUL_RES).send({'error': error}).end();
+    });
+  }).catch((error) => {
+    return res.status(UNSUCCESSFUL_RES).send({'error': error}).end();
+  });
+});
+
+// Updates an account's disabled/enabled status
+app.post('/updateAccountStatus', async (req,res) => { 
+  res = setResAttr(res,'*','POST','Content-Type, application/json');
+
+  // Get info from req
+  const reqInfo = {
+    'body': req.body,
+    'uid': req.body.uid,
+    'token': req.body.adminToken
+  };
+
+  // Set the user disabled status
+  const setDisabled = async (disValue) => {
+    await admin.auth().updateUser(reqInfo['uid'], {
+      disabled: disValue
+    });
+  };
+
+  // Check for valid user token
+  admin.auth(adminApp).verifyIdToken(req.body.token).then((decodedToken) => {
+    admin.auth(adminApp).getUser(decodedToken.uid).then((userRecord) => {
+      let claims = userRecord.customClaims;
+      if(claims !== null && (ADMIN_STR in claims && MANAGER_STR in claims)) {
```

Comment:
It is to double check that the request came from a RRDB user and not just one from the whole database. It just checks that the admin and manager claims are present within the user claims. It may not be needed in production.

Reply:
Ah that makes sense.